src/libs/ReportUtils.js

```diff
@@ -3316,11 +3342,66 @@ function canAccessReport(report, policies, betas, allReportActions) {
  */
 function shouldHideReport(report, currentReportId) {
     const parentReport = getParentReport(getReport(currentReportId));
-    const reportActions = ReportActionsUtils.getAllReportActions(report.reportID);
-    const isChildReportHasComment = _.some(reportActions, (reportAction) => (reportAction.childVisibleActionCount || 0) > 0);
+    const allReportActions = ReportActionsUtils.getAllReportActions(report.reportID);
+    const isChildReportHasComment = _.some(allReportActions, (reportAction) => (reportAction.childVisibleActionCount || 0) > 0);
     return parentReport.reportID !== report.reportID && !isChildReportHasComment;
 }
 
+/**
+ * @param {String} transactionID
+ * @returns {Boolean}
+ */
+
+function transactionHasViolation(transactionID) {
+    const violations = lodashGet(transactionViolations, transactionID, []);
```

Comment:
@cead22 I was able to test this out by altering this to:
```
const fakeTransactionViolations = {
        [transactionID]: [{type: 'violation'}, {type: 'otherType'}],
        ...otherTransactions,
    };

    const violations = lodashGet(fakeTransactionViolations, transactionID, []);
```

Using this I was able to test out `transactionThreadHasViolations` and `reportHasViolations`. Are the backend changes close enough to finishing that we can hold this for review until then?

Reply:
No not really, but I think we should merge, and when the backend changes are ready we'll test, and if there are bugs we can create new issues. This is behind a beta anyway

Reply:
@cead22 okay, sounds good. I'll get this ready for review then.

Reply:
Thanks! Ping me when it's ready